CriticalMass/MapViewController.swift

```diff
@@ -181,35 +178,61 @@ class MapViewController: UIViewController {
         mapView.addOverlay(nightThemeOverlay, level: .aboveRoads)
     }
 
-    @objc func didReceiveInitialLocation(notification: Notification) {
-        guard let location = notification.object as? Location else { return }
-        focusOnLocation(location: location)
-
-        // TODO: Replace test implemenation with controller based
-        guard Feature.events.isActive else { return }
-        let coordinate = CLLocationCoordinate2D(
-            latitude: location[keyPath: \Location.latitude],
-            longitude: location[keyPath: \Location.longitude]
-        )
-        nextRideHandler.getNextRide(around: coordinate) { result in
+    private func getNextRide(_ coordinate: CLLocationCoordinate2D) {
```

Comment:
We should move that ride logic to the AnnotationController. The idea behind AnnotationController is that it contains or controls all the logic that is need to display annotations

Reply:
But this call also triggers the MapInfoView

Reply:
I didn't really get the AnnotationController tbh. I moved the Annotation related logic into a AnnotationController but also having the API call which has other effects into that and then reaching out again does not make sense to me

Reply:
I would like to keep the MapViewController as stupid as possible because I'm worried that it might become a "Godclass" at some point. 
The idea is that the MapViewController is just just contains logic to communicate between the AnnotationController and the Map. The Content related logic (Bikes, Events and Friends) should be in specific AnnotationController which makes it also easier to test.

To workaround that specific need, you could inject the MapInfoViewController into the AnnotationController. But I'm also okay with keeping it as it is for now and doing that in a Future PR

Reply:
Like that idea. If a future PR is okay then I'll go with that option 🏗

Reply:
Having references to MapViewController properties in these controllers makes it harder to reason about these I think since they have shared state. Maybe having one instead of one for each content type. But not making the Map a god makes sense for sure